include/boost/geometry/strategies/cartesian/intersection.hpp

```diff
@@ -176,36 +176,43 @@ struct cartesian_segments
             }
         }
 
-    public :
-        template <typename Point, typename Segment1, typename Segment2>
-        void calculate(Point& point, Segment1 const& a, Segment2 const& b) const
+        bool use_a() const
         {
-            bool use_a = true;
-
-            // Prefer one segment if one is on or near an endpoint
-            bool const a_near_end = robust_ra.near_end();
-            bool const b_near_end = robust_rb.near_end();
-            if (a_near_end && ! b_near_end)
-            {
-                use_a = true;
-            }
-            else if (b_near_end && ! a_near_end)
+            // Prefer the segment closer to the endpoint.
+            // If both are about equally close, then prefer the longer segment
+            // To avoid hard thresholds, behavior is made fluent:
+            // Calculate comparable length indications,
+            // the longer the segment (relatively), the lower the value
+            // such that the shorter lengths are evaluated higher and will
+            // be preferred.
+
+            auto const cla = comparable_length_a();
+            auto const clb = comparable_length_b();
+            auto const clm = std::max(cla, clb);
```

Comment:
This line fails with msvc. It should've been:
```
auto const clm = (std::max)(cla, clb);
```

Reply:
:heavy_check_mark: